firestore.rules

```diff
@@ -16,7 +11,22 @@ service cloud.firestore {
       resource.data.authorId == request.auth.uid);
     }
 
-    // Add new rules here
+    //Allow authenticated user to view, create, update and delete their own todos
+
+    match /todos/{todoId} {
+      allow read: if request.auth.token.email_verified;
+      allow write: if request.auth.uid == request.resource.data.uid;
+      allow delete: if request.auth.uid == resource.data.uid;
+    }
+
+    //Allow any authenticated user to view other users profiles, edit or create a profile
+    //Allow ONLY Admins to edit ANY users Profile
+
+    match /users/{userId} {
+      allow read: if request.auth.token.email_verified;
+      allow write: if request.auth.uid == request.resource.data.uid;
+      allow update: if get(/databases/$(database)/documents/users/$(request.auth.uid)).data.isAdmin == true;
```

Comment:
As mentioned above, `update` is a part of `write`, so writing the rules in this way creates overlapping conditions that may be confusing. The rules above allow non-admin users to change admin status because a user is allowed to perform all `write` operations on their own profile. The following rule doesn't cancel it or narrow it down. That is why having a single rule for every operation is better. An update rule that doesn't allow changing the admin status unless edited by an admin would look something like this:
```
allow update: if request.auth != null 
      && (
        request.resource.data.isAdmin == resource.data.isAdmin
        || get(/databases/$(database)/documents/users/$(request.auth.uid)).data.isAdmin == true
      )
      && (
        request.auth.uid == resource.data.uid 
        || get(/databases/$(database)/documents/users/$(request.auth.uid)).data.isAdmin == true
      );
      ```